testdata/p4_16_samples_outputs/stack-frontend.p4

```diff
@@ -5,26 +5,23 @@ header h {
 
 parser p() {
     @name("p.stack") h[4] stack_0;
-    @name("p.b") h b_0;
     state start {
         stack_0[0].setInvalid();
         stack_0[1].setInvalid();
         stack_0[2].setInvalid();
         stack_0[3].setInvalid();
         stack_0[3].setValid();
-        b_0 = stack_0.last;
-        stack_0[2] = b_0;
```

Comment:
Why is this statement eliminated? We have to make sure this is not a bug.

Reply:
I checked this out and it looks like this happens because the header stack is unused overall. Attached is the same example but with output that matters. In this case the assignment is not removed. Concerning to the comment above, the `set(In)Valid` calls could also be removed here? Btw Gauntlet does not pick up changes in dead code. This can be a blind spot for some of these tests.

Modified `stack.p4`: 
```p4
#include <core.p4>

header H {
    bit<8> a;
}


struct Headers {
    H[4] stack;
}

parser p(packet_in pkt, out Headers h) {
    state start {
        H[4] stack;
        stack[3].setValid();
        H b = stack[3];
        b = stack.last;
        stack[2] = b;
        b = stack.next;
        bit<32> e = stack.lastIndex;
        h.stack = stack;
        transition accept;
    }
}

control ingress(inout Headers h) {
    apply {
        H[4] stack;
        stack[3].setValid();
        H b = stack[3];
        stack[2] = b;
        stack.push_front(2);
        stack.pop_front(2);
        bit<32> sz = stack.size;
        h.stack = stack;
    }
}

parser Parser(packet_in b, out Headers hdr);
control Ingress(inout Headers hdr);
package top(Parser p, Ingress ig);
top(p(), ingress()) main;
```

Reply:
Calls like setInvalid have side-effects, and we do not have any analysis which would remove calls with side-effects. We do not understand currently the fact that the side-effects are confined just to an unused variable.